src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -117,6 +132,31 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
         blockExplorerUriTemplate,
       }
     },
+    onSetSafeSettings: (settings: SafeSettings) => {
+      const EIP_1271_SUPPORTED_SAFE_VERSION = '1.0.0'
+
+      if (typeof settings.offChainSigning === 'boolean' && gte(safe.version, EIP_1271_SUPPORTED_SAFE_VERSION)) {
+        setOffChainSigning(settings.offChainSigning)
+      }
+
+      return true
+    },
+    onGetOffChainSignature: async (messageHash: string) => {
+      const safeMessage = safeMessages.data?.results
+        ?.filter(isSafeMessageListItem)
+        ?.find((item) => item.messageHash === messageHash)
+
+      if (safeMessage) {
+        return safeMessage.preparedSignature
+      }
+
+      try {
+        const { preparedSignature } = await getSafeMessage(chainId, messageHash)
+        return preparedSignature
+      } catch {
+        // ignore
+      }
+    },
```

Comment:
The store only polls the first page of messages. If the signature is on page 2, for example, it will not be found locally, hence requiring the `fetch`.

Reply:
Ok, that makes sense so its more of an optimization to check the store before fetching. I would suggest leaving a comment as it could be confusing as to why we do both.

Reply:
we should return something here in the `catch` because if it fails the sdk call will be pending right?